src/uPlot.js

```diff
@@ -1761,21 +1816,38 @@ export default function uPlot(opts, data, then) {
 		}
 	}
 
-	function mouseUp(e, src, _x, _y, _w, _h, _i) {
-		if ((e == null || filtMouse(e))) {
+	function mouseUp(e, src, _x, _y, _w, _h, _i, _drag) {
+		const isSyncReq = src != null;
+		let shouldDrag;
+
+		if (isSyncReq || filtMouse(e)) {
 			dragging = false;
 
 			cacheMouse(e, src, _x, _y, _w, _h, _i, false, true);
 
-			if (mouseLeft1 != mouseLeft0 || mouseTop1 != mouseTop0) {
+			let locked = cursor.lock && cursor.locked;
+
+			if (_drag == null) {
+				let shouldXDrag = abs(mouseLeft0 - mouseLeft1) / plotWidCss > 0.002;
```

Comment:
this whole new block scares me, especially with hard-coded thresholds. was the original logic of `mouseLeft1 != mouseLeft0 || mouseTop1 != mouseTop0` no longer sufficient to determine if a mousemove happened?

or is this guarding against the case where the synced scales are so different that a mousemove along one chart is insufficient to move 1px on another?

Reply:
This is needed when you are determining drag inside of mouseup instead of at mousemove, no longer needed (though I was thinking select[WIDTH] and select[HEIGHT] are the clearer things to check since that's what ends up being used in all the functions)

Reply:
Okay this is the one that causes a cursor lock issue when you remove it. But I think it is because the select isn't being cleared. So this is a bit of a hack to compensate for something else. Investigating more...